### PR TITLE
dump/load method in mp.py to allow dump and load Mpo and refine SineDVR basis

### PR DESCRIPTION
**Pull request by jjren**

### CODE REVIEW

File: renormalizer/mps/mp.py [hunk 1]
```diff
@@ -934,6 +934,38 @@ def _array2mt(self, array, idx, allow_dump=True):
     
     def build_empty_mp(self, num):
         self._mp = [[None]] * num
+    
+    def dump(self, fname, other_attrs=[]):
```
Comment (liwt31) [inline renormalizer/mps/mp.py:938]: Not a big fan of using `[]` as the default argument (see [this post](https://www.toptal.com/python/top-10-mistakes-that-python-programmers-make))

File: renormalizer/model/basis.py [hunk 2]
```diff
@@ -382,45 +383,253 @@ def op_mat(self, op: Union[Op, str]):
         if op_symbol == "I":
             mat = np.eye(self.nbas)
         
-        elif op_symbol.split("^")[0] == "x":
+        elif op_symbol.split("^")[0] == "x" and " " not in op_symbol:
```
Comment (liwt31) [inline renormalizer/model/basis.py:386]: Is these code just for checking or it does something different from the good old `mat = np.diag(self.dvr_x ** moment)`?

**Comment by liwt31** [comment]
This commit seems to involve a large number of equations. Are they derived or from literature? If it's the latter case, adding the reference to the document could be helping.

**Comment by jjren** [comment]
The matrix element is derived through some manual integrations.

File: renormalizer/model/basis.py (hunk 2 above)
Comment (jjren) [reply renormalizer/model/basis.py:386]: The modification ensures analytical accuracy for the operators such as `x p`, `x^2`, `x^3`, `x p^2` etc.

Comment (liwt31) [reply renormalizer/model/basis.py:386]: Am I to understood that even for `x` `mat = np.diag(self.dvr_x)` is not accurate?

Comment (jjren) [reply renormalizer/model/basis.py:386]: For `x`, it is exact. But for `x^2`, `x^3`, etc, it is not.

File: renormalizer/mps/mp.py (hunk 1 above)
Comment (jjren) [reply renormalizer/mps/mp.py]: fixed

**Review by liwt31 — APPROVED** [review]

**Pull request merged**